snaps/features/custom-ui/user-defined-components.md

```diff
@@ -0,0 +1,192 @@
+---
+description: Create your own JSX components to improve readability.
+sidebar_position: 5
+---
+
+# User-defined components
+
+When using JSX, you can create your own components by composing [existing components](with-jsx.md), or
+other user-defined components.
+
+## Basic example
+
+In this first, basic example, the user-defined component is static. It does not accept any props (parameters) and returns the contents of a static home page.
+
+```jsx title="Home.jsx"
+import { Box, Heading, Text } from "@metamask/snaps-sdk/jsx";
+
+export const Home = () => {
+  return (
+    <Box>
+      <Heading>Welcome to my Snap</Heading>
+      <Text>Hello, world!</Text>
+    </Box>
+  );
+};
+```
+
+Once the component is defined, it can be used anywhere in the Snap. For example, to display the home page, you can use the following code:
+
+```jsx title="index.jsx"
+import { Home } from "./Home";
+
+export const onHomepage = () => {
+  return <Home />;
+};
+```
+
+## Example with props
+
+Components can be parametrized by passing props. Props are passed to the component as an object and can be accessed using the first parameter of the component's definition function:
+
+```jsx title="Insight.jsx"
+export const Insight = (props) => {
+  return (
+    <Box>
+      <Row label="From">
+        <Address address={props.from} />
+      </Row>
+      <Row label="To">
+        {to ? <Address address={props.to} /> : <Text>None</Text>}
+      </Row>
+    </Box>
+  );
+};
+```
+
+In the example above, we see two usages of props:
+
+- The `Insight` component accepts a `props` parameter, which is an object containing the `from` and `to` addresses. The `from` address is accessed using `props.from`, and the `to` address is accessed using `props.to`, since `props` is just a regular JavaScript object.
+- The `Insight` component then uses the built-in `Address` component to display addresses. The `Address` component accepts an `address` prop. When using the `Address` component, we pass props to it by using a notation similar to HTML attributes: `address={props.from}`.
+
+To use the `Insight` component, you can pass the `from` and `to` addresses as props:
+
+```jsx title="index.jsx"
+import { Insight } from "./Insight";
+
+export const onTransaction = () => {
+  return { content: <Insight from="0x123" to="0x456" /> };
```

Comment:
This is really good but I am pretty sure this specific example will fail because Address component expects the address value to be the correct length and valid

Reply:
fixed